src/city/population.c

```diff
@@ -317,6 +342,83 @@ static void yearly_recalculate_population(void)
     city_data.population.average_per_year = city_data.population.total_all_years / city_data.population.total_years;
 }
 
+int calculate_total_housing_buildings(void)
+{
+    int total = 0;
+    for (int i = 1; i < MAX_BUILDINGS; i++) {
+        building *b = building_get(i);
+        if (b->state == BUILDING_STATE_UNUSED ||
+            b->state == BUILDING_STATE_UNDO ||
+            b->state == BUILDING_STATE_DELETED_BY_GAME ||
+            b->state == BUILDING_STATE_DELETED_BY_PLAYER) {
+            continue;
+        }
+        if (building_is_house(b->type) && b->house_population > 0) {
+            total += 1;
+        }
+    }
+
+    return total;
+}
+
+int * calculate_number_of_each_housing_type(void) {
+
+   static int housing_type_counts[20] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+
+    for (int i = 0; i <= 19; i++) {
+        housing_type_counts[i] = 0;
+    }
+
+    for (int i = 1; i < MAX_BUILDINGS; i++) {
+        building *b = building_get(i);
+        if (b->state == BUILDING_STATE_UNUSED ||
+            b->state == BUILDING_STATE_UNDO ||
+            b->state == BUILDING_STATE_DELETED_BY_GAME ||
+            b->state == BUILDING_STATE_DELETED_BY_PLAYER) {
+            continue;
+        }
+        if (b->house_size) {
+            housing_type_counts[b->subtype.house_level] +=1;
+        }
+    }
+
+   return housing_type_counts;
+}
+
+int * calculate_houses_demanding_goods(int * housing_type_counts) {
+    static int houses_demanding_goods[4] = {0, 0, 0, 0};
+
+    for (int i = 0; i <= 3; i++) {
```

Comment:
Better to use model_get_house to check for requirements instead of hard coding it, in case someone edits the model file.